### PR TITLE
Update exact_syn_from_label.ru

### PR DESCRIPTION
**Pull request by matentzn**
Resolves #ISSUE(s).

## Overview

This PR:
- One
- Two
- Three

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [ ] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [ ] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary
   
### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [ ] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [ ] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [ ] Yes

### CODE REVIEW

**Review by twhetzel — CHANGES_REQUESTED** [review]
a test of the components make goal, e.g. `sh run.sh make components/doid.owl -B` now works following Joe's update, but I am not seeing the `GENERATED_FROM_LABEL` annotation.

@joeflack4 let me know if you are looking into this or I should take this. thanks

File: src/sparql/exact_syn_from_label.ru [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Create an exact synonym copy for all labels
 PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
 PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
```
[the remark this answers is not in the document]
Comment (joeflack4) [reply src/sparql/exact_syn_from_label.ru:4]: @twhetzel Wow you got onto that fast. Yes, I avoid them quite often these days. I did that last one within 1 minute, sorry you encountered it.

**Review by joeflack4 — COMMENTED** [review]
x

File: src/sparql/exact_syn_from_label.ru
Comment (joeflack4) [inline src/sparql/exact_syn_from_label.ru:1]: ## General review
Good thinking w/ the new annotation prop, Nico.

I ran this locally and it works fine.

@twhetzel I think we should merge this first so that we can use it in #783, without running a build. Or, probably even better, I'll just effectively branch #783 off of this, and we can skip the build for this one and just do the build on #783 instead.

FYI changed the base to `develop`.

I also pushed 2 new commits to finish some issues with this (missing rdf namespace & properties.txt entry).

@matentzn How do you intend this to work alongside the `MONDO:GENERATED` created by https://github.com/monarch-initiative/mondo-ingest/blob/main/src/sparql/fix-labels-with-brackets.ru? I suppose that is a bad file name, because it doesn't fix labels. It creates synonyms from labels and other fields. And it also will create them whether or not there are brackets. It will just fix/remove brackets if they exist. Here's how the props are coming out now:


```owl
<owl:Class rdf:about="http://purl.obolibrary.org/obo/DOID_7086">
    <rdfs:label xml:lang="en">multicentric papillary thyroid carcinoma</rdfs:label>
</owl:Class>
<owl:Axiom>
    <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/DOID_7086"/>
    <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
    <owl:annotatedTarget xml:lang="en">multicentric papillary thyroid carcinoma</owl:annotatedTarget>
    <oboInOwl:hasSynonymType rdf:resource="http://purl.obolibrary.org/obo/mondo#GENERATED_FROM_LABEL"/>
</owl:Axiom>
<owl:Axiom>
    <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/DOID_7086"/>
    <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
    <owl:annotatedTarget>multifocal papillary thyroid carcinoma</owl:annotatedTarget>
    <oboInOwl:hasSynonymType rdf:resource="http://purl.obolibrary.org/obo/mondo#GENERATED"/>
</owl:Axiom>
```

**Review by joeflack4 — APPROVED** [review]

File: src/sparql/exact_syn_from_label.ru
Comment (twhetzel) [reply src/sparql/exact_syn_from_label.ru:1]: Thanks for pushing the "properties.txt" file, I am now getting the `#GENERATED_FROM_LABEL` tags.

Comment (twhetzel) [reply src/sparql/exact_syn_from_label.ru:1]: @joeflack4 what is your concern with the other query, `[fix-labels-with-brackets.ru](https://github.com/monarch-initiative/mondo-ingest/blob/main/src/sparql/fix-labels-with-brackets.ru)` and the query in this PR?

Comment (twhetzel) [reply src/sparql/exact_syn_from_label.ru:1]: Based on your comment, one concern is that this query is not using the already "fixed" label when the original label contains brackets. For example, for http://id.who.int/icd/entity/186800184 'Metastatic nasopharyngeal carcinoma [primary nasopharyngeal carcinoma spreading elsewhere]' in ICD11 here's what happens after running `sh run.sh make components/icd11foundation.owl -B` on this branch
Screenshot 2025-02-19 at 1 58 49 PM


And in these cases it would be better that the `#GENERATED_FROM_LABEL` exact synonym is created from the `#GENERATED` exact synonym. I am running a test to check another option for the query in this PR....

Comment (joeflack4) [reply src/sparql/exact_syn_from_label.ru:1]: Yeah, my concern was that adding `#GENERATED` as well feels a little redundant when there's already `#GENERATED_FROM_LABEL`. Though it's not a big gripe. At this current time, I'm not actually opposed to completing the work quickly as opposed to getting it perfect stylistically and such.

But @twhetzel you bring up a good point, that the `#GENERATED_FROM_LABEL` would be better done after the brackets fixing.

Comment (joeflack4) [reply src/sparql/exact_syn_from_label.ru:1]: @twhetzel You asked what my other general gripes are with [fix-labels-with-brackets.ru](https://github.com/monarch-initiative/mondo-ingest/blob/main/src/sparql/fix-labels-with-brackets.ru). That is explained in [this comment](https://github.com/monarch-initiative/mondo-ingest/pull/785#discussion_r1962379284); not within scope of this issue.

File: src/sparql/exact_syn_from_label.ru [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+    owl:annotatedSource ?cls ;
+    owl:annotatedProperty oboInOwl:hasExactSynonym ;
+    owl:annotatedTarget ?cleanLabel ;
+    oboInOwl:hasSynonymType <http://purl.obolibrary.org/obo/mondo#GENERATED_FROM_LABEL>
+  ] .
+} 
+WHERE {
   ?cls rdfs:label ?label .
+
+  BIND(REPLACE(STR(?label), " [\\[(].*[)\\]]$", "") AS ?cleanLabel)
```
Comment (twhetzel) [inline src/sparql/exact_syn_from_label.ru:26]: I tested this update with ICD11 and here's an example IRI, http://id.who.int/icd/entity/1945176477, where it's using the "cleaned" label to convert to the exact synonym tagged as `GENERATED_FROM_LABEL`. Tested by running: `sh run.sh make components/icd11foundation.owl -B`

Comment (twhetzel) [reply src/sparql/exact_syn_from_label.ru:26]: @matentzn are you ok with this addition to the query or do you prefer the original label be used as the `GENERATED_FROM_LABEL` exact synonym before any attempts to "fix" the label due to the presence of brackets? 

One question I have is how well this "fix" for labels was previously tested and reviewed to confirm it's working as expected in all cases so that we do not add badly formatted synonyms into Mondo.

Comment (matentzn) [reply src/sparql/exact_syn_from_label.ru:26]: I am not sure. This fix was needed for ICD10, and was not really tested with anything else. I am not sure this is a good idea to make default for _all_ ontologies. I would do this only for ICD11 and ICD10. There are some legitimate reasons of having brackets in class names..

Comment (matentzn) [reply src/sparql/exact_syn_from_label.ru:26]: Maybe I would just leave the unchanged label for now.. for safety reasons

Comment (twhetzel) [reply src/sparql/exact_syn_from_label.ru:26]: Sure, while I would like to have more meaningful synonyms, ie without brackets etc., if this bracket "fix" was not more deeply tested then we run the risk of adding even worse formatted synonyms into Mondo which is a big negative. I will revert this change, but this small though exercise has been useful.

File: src/sparql/exact_syn_from_label.ru
Comment (twhetzel) [reply src/sparql/exact_syn_from_label.ru:1]: FYI - the `#GENERATED_FROM_LABEL` is used since the "fix" brackets query works on both labels and synonyms so this new tag makes it explicit where this new synonym is from as well as the parallel discussion in this PR about that the "fix" for brackets should not be applied to labels being used as synonyms at this time.

**Review by twhetzel — APPROVED** [review]
lgtm

**Comment by twhetzel** [comment]
For this specific PR, I am ok with merging this into `develop` before doing a full data build so that Joe can move forward with  the other issues to work on. I will do a feature branch data build, however as a sanity check.
